src/msw/webview_ie.cpp

```diff
@@ -853,25 +857,138 @@ wxString wxWebViewIE::GetPageText() const
     }
 }
 
-void wxWebViewIE::RunScript(const wxString& javascript)
+bool wxWebViewIE::RunScript(const wxString& javascript, wxString* output)
 {
     wxCOMPtr<IHTMLDocument2> document(GetDocument());
+    IDispatch* scriptDispatch = NULL;
 
-    if(document)
+    if ( !document )
     {
-        wxCOMPtr<IHTMLWindow2> window;
-        wxString language = "javascript";
-        HRESULT hr = document->get_parentWindow(&window);
-        if(SUCCEEDED(hr))
+        wxLogWarning(_("HTML document is null"));
+        return false;
+    }
+
+    if ( FAILED(document->get_Script(&scriptDispatch)) )
+    {
+        wxLogWarning(_("Can't get the Javascript"));
+        return false;
+    }
+
+    wxString counter;
+    wxString javaScriptVariable =
+        wxWebViewUtils::WrapJavaScript(javascript,
+        &m_runScriptCount, &counter);
+
+    wxAutomationObject scriptAO(scriptDispatch);
+    wxVariant varJavascript(javaScriptVariable);
+    wxVariant varResult;
+
+    if ( !scriptAO.Invoke("eval", DISPATCH_METHOD, varResult, 1, &varJavascript) )
+    {
+        wxLogWarning(_("Can't run Javascript"));
+        return false;
+    }
+
+    if ( varResult.IsType("bool") && varResult.GetBool() )
+    {
+        varJavascript = "(__wx$" + counter + " == null || \
+                        typeof __wx$" + counter + " != 'object') ? \
+                        String(__wx$" + counter + ") : __wx$" + counter;
+        if ( !scriptAO.Invoke("eval", DISPATCH_METHOD, varResult, 1, 
+            &varJavascript) )
         {
-            VARIANT level;
-            VariantInit(&level);
-            V_VT(&level) = VT_EMPTY;
-            window->execScript(wxBasicString(javascript),
-                               wxBasicString(language),
-                               &level);
+            wxLogWarning(_("Can't run Javascript"));
+            return false;
+        }
+
+        if ( varResult.IsType("void*") )
+        {   // Script returned an Object (JScriptTypeInfo IDispatch), convert it to JSON
+            IDispatch* dispatchResult = (IDispatch*)varResult.GetVoidPtr();
+            wxAutomationObject JSONAO;
+
+            if ( dispatchResult )
+            {
+                // JSON is not available in Quirks or IE6/7 standards mode,
+                // which is unfortunately the default one for the embedded
+                // browser control, see
+                // https://docs.microsoft.com/en-us/scripting/javascript/reference/json-object-javascript#requirements
+                // and see here how to make a program run use "modern" modes
+                // https://msdn.microsoft.com/en-us/library/ee330730(v=vs.85)#browser_emulation
+                if (scriptAO.GetObject(JSONAO, "JSON"))
```

Comment:
Tiny style issue: There should be spaces around the expression inside parentheses.